### PR TITLE
add checksum extension

### PR DESCRIPTION
**Pull request by Acconut**

### CODE REVIEW

**Comment by kvz** [comment]
LGTM

File: protocol.md [hunk 1]
```diff
@@ -317,8 +317,47 @@ The value of the  `Upload-Expires` header MUST be in
 
 ### Checksums
 
-This extension will define how to provide per file or per chunk checksums for
-uploaded files.
+Clients and servers MAY implement and use this extension to verify data
+integrity per chunk. In this case the server MUST add the `checksum` element to
+the `TUS-Extension` header.
+
+A client MAY include the `Content-MD5` header and its appropriate value in a
```
Comment (qsorix) [inline protocol.md]: Refer to RFC for what is meant by "proper value"? It's actually quite tricky, IMHO, and may result in incompatible implementations. People may skip the example and assume it should be the hex-string with the checksum. Others may use base64 of that. Content-Transfer-Encoding needs to be taken into consideration, etc.

The rest looks great.

Comment (kvz) [inline protocol.md]: What would be your (rough, just to get an idea what you would like to see changed/included) proposal?

Comment (qsorix) [inline protocol.md]: I think I would just say that RFC 2616 defines how to compute and represent the checksum and drop a link to http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.15. Just as it's done already regarding trailers.

Comment (kvz) [inline protocol.md]: Sounds like a good idea

Comment (Acconut) [inline protocol.md]: Good idea, added in d798dfb.

**Pull request merged**